### PR TITLE
feat(metrics): Add user_misery to derived metrics

### PR DESCRIPTION
**Pull request by ahmedetefy**
Adds user_misery to list of derived metrics
and exposes it in naming layer so it is
queryable

### CODE REVIEW

File: src/sentry/snuba/metrics/naming_layer/mri.py [hunk 1]
```diff
@@ -83,3 +83,5 @@ class TransactionMRI(Enum):
     TOLERATED = "e:transactions/tolerated@none"
     APDEX = "e:transactions/apdex@ratio"
     MISERABLE_USER = "e:transactions/user.miserable@none"
+    ALL_USER = "e:transactions/user.all@none"
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/naming_layer/mri.py:86]: nit: For sessions, we have `ERRORED_USER_ALL`. Should we call this one `USER_ALL`?

File: src/sentry/snuba/metrics/fields/base.py [hunk 2]
```diff
@@ -953,6 +958,22 @@ def run_post_query_function(
...
+            snql=lambda *_, org_id, metric_ids, alias=None: uniq_aggregation_on_metric(
+                metric_ids, alias=alias
+            ),
+        ),
+        SingularEntityDerivedMetric(
+            metric_mri=TransactionMRI.USER_MISERY.value,
+            metrics=[TransactionMRI.MISERABLE_USER.value, TransactionMRI.ALL_USER.value],
+            unit="percentage",
+            snql=lambda miserable_user, user, org_id, metric_ids, alias=None: division_float(
+                addition(miserable_user, MISERY_ALPHA), addition(user, MISERY_BETA), alias
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/fields/base.py:974]: If I read the discover code correctly, this should be

```suggestion
                addition(miserable_user, MISERY_ALPHA), addition(user, MISERY_ALPHA + MISERY_BETA), alias
```

reference:

https://github.com/getsentry/sentry/blob/23d11f0077bc4ed617c7b170074d1f6e2e4fb612/src/sentry/search/events/datasets/discover.py#L1202

https://github.com/getsentry/sentry/blob/23d11f0077bc4ed617c7b170074d1f6e2e4fb612/src/sentry/search/events/datasets/discover.py#L211

File: src/sentry/snuba/metrics/naming_layer/mri.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/naming_layer/mri.py:86]: I am following this https://github.com/getsentry/sentry/blob/5e9a29d87613d21c8e12403972b15bcb88538cc0/src/sentry/snuba/metrics/naming_layer/mri.py#L40

`ERRORED_USER_ALL` is just an internal name added for to simplify computation, and I chose this name for lack of better naming but I'd say if we want to be consistent we should probably follow the link I referenced

File: src/sentry/snuba/metrics/fields/base.py (hunk 2 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/fields/base.py:974]: Good catch! will fix

File: src/sentry/snuba/metrics/naming_layer/mri.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/snuba/metrics/naming_layer/mri.py:88]: OK makes sense.

**Review by jjbayer — APPROVED** [review]

**Pull request merged**